src/tablist.js

```diff
@@ -18,6 +18,10 @@ SideTabList.prototype = {
     if (this.alwaysShrink) {
       this.maybeShrinkTabs();
     }
+
+    this.useRegExpFilter = (await browser.storage.local.get({
```

Comment:
What I'm saying is that you could rename _detectDarkTheme to _readOptions (or a better name if you can think of one), add your regex filter option detection there so all options are read in the background without delaying the construction of the UI. (it's not like we need them right away)